python/ray/data/dataset.py

```diff
@@ -536,10 +536,8 @@ def transform(
             output_buffer = BlockOutputBuffer(None, context.target_max_block_size)
             # Ensure that zero-copy batch views are copied so mutating UDFs don't error.
             batcher = Batcher(batch_size, ensure_copy=batch_size is not None)
-            for block in blocks:
-                batcher.add(block)
-            batcher.done_adding()
-            while batcher.has_any():
+
+            def process_next_batch() -> Iterator[Block]:
```

Comment:
For readability, it'd be clear to pass in batch as arg, so it read like process_next_batch(batch) when calling.